src/Aspire.Hosting/MySql/MySqlDatabaseResource.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents a MySQL database. This is a child resource of a <see cref="MySqlContainerResource"/>.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+/// <param name="mySqlContainer">The MySQL server resource associated with this database.</param>
+public class MySqlDatabaseResource(string name, MySqlContainerResource mySqlContainer) : Resource(name), IMySqlResource, IResourceWithParent<MySqlContainerResource>
+{
+    public MySqlContainerResource Parent { get; } = mySqlContainer;
+
+    /// <summary>
+    /// Gets the connection string for the MySQL database.
+    /// </summary>
+    /// <returns>A connection string for the MySQL database.</returns>
+    public string? GetConnectionString()
+    {
+        if (Parent.GetConnectionString() is { } connectionString)
+        {
+            return $"{connectionString}Database={Name}";
+        }
+        else
+        {
+            throw new DistributedApplicationException("Parent resource connection string was null.");
```

Comment:
@eerhardt should we start moving these things to a resx? I assume we won't localize unless the base libraries start to localize. So perhaps there isn't much value.

Reply:
ASP.NET Core doesn't have a lot of its exception messages in resx's either. For example

https://github.com/dotnet/aspnetcore/blob/f8f03ea3764826d009f7ceb9bb91e482ce4a3fa9/src/Shared/Json/JsonSerializerExtensions.cs#L26

If we ever needed to localize our exceptions, it wouldn't be that hard to find these. But odds are, after 8 years of .NET Core System.* libraries not being localized, I'd be surprised if it ever happens.

Reply:
Agreed.